spec/system/managing_topics_spec.rb

```diff
@@ -27,4 +27,12 @@
     expect(current_path).to eq topic_path(topic)
     expect(page).to have_content('Topic:')
   end
+
+  scenario 'when deleting a topic' do
+    visit topic_path(topic)
+    # expect {click_on 'Delete Topic'}.to change(Topic, :count).by(-1)
```

Comment:
Commented out code? If it doesn't work then remove. Don't leave commented code.
Were you wondering how to combine it with `accept_confirm`?
This should work:
```
expect {
  accept_confirm { click_on 'Delete Topic' }
}.to change(Topic, :count).by(-1)
```
Alternatively, us

Reply:
Yeah... I had tried something like that and couldn't get it working... just tried your example and it doesn't work either.... "expected -1 and was changed by 0" is the result :/

Reply:
will leave the commented out version in until I catch up with you.

Reply:
This isn't majorly important, so you can just leave this.

However, if you want to persevere, here are some thoughts about further things to try (or think about):

Throw in a `sleep x` statement. The thinking here is that the `accept_confirm` Capybara method is actually working by executing some javascript on the page. It may however be returning immediately (i.e. before the javascript has finished executing), and thus triggering RSpec's checks too soon. Adding a sleep puts the process to sleep for a specified number of seconds before continuing
```ruby
expect {
  accept_confirm { click_on 'Delete Topic' }
  sleep 3 # puts the process to sleep for 3 seconds
}.to change(Topic, :count).by(-1)
```

Generally sleeps are a bad sign; instead we should make assertions that will cause Capybara to wait until they are fulfilled. So in this case you could just move your next expect statement up to cause Capybara (and therefore RSpec) to wait until the page change has happened before assessing whether the Topic count has changed.

```ruby
expect {
  accept_confirm { click_on 'Delete Topic' }
  expect(current_path).to eq topics_path
}.to change(Topic, :count).by(-1)
```

Another thing to try is using a block on both ends of the RSpec statement
```ruby
expect {
  accept_confirm { click_on 'Delete Topic' }
  expect(current_path).to eq topics_path
}.to change { Topic.count) }.by(-1)
```

Another way we often handle Capybara and Javascript is adding our own asynchronous wait methods. Here's a link that explores this: https://gist.github.com/metaskills/1172519